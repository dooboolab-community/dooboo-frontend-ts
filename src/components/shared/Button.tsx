import React, { Component } from 'react';
import styled from 'styled-components';
import { WhiteButton, TransparentButton } from '../ui/Buttons';

interface IProps {
  id?: string;
  white?: boolean;
  imgSrc?: any;
  text?: string;
  onClick?: () => void;
  isLoading?: boolean;
}

const Text = styled.span`
  font-size: 14px;
  color: rgb(128, 109, 216);
`;

const LogoImg = styled.img`
  position: absolute;
  left: 16px;
  height: 20px;
  width: 20px;
  object-fit: cover
`;

const Spinner = styled.div`
  border: 4px solid #f3f3f3; /* Light grey */
  border-top: 4px solid #3498db; /* Blue */
  border-radius: 50%;
  width: 24px;
  height: 24px;
  animation: spin 1s linear infinite;
  @keyframes spin {
    0% { transform: rotate(0deg); }
    100% { transform: rotate(360deg); }
  }
`;

function Button(props: IProps) {
  const { white, onClick, imgSrc, text } = props;
  if (props.white) {
    return (
      <WhiteButton
        style={{ height: '60px' }}
        onClick={onClick}
      >
        {
          props.isLoading
            ? <Spinner id='spinner'/>
            : <div>
              {
                props.imgSrc
                  ? <LogoImg
                    src={props.imgSrc}
                  />
                  : null
              }
              <Text>{props.text}</Text>
            </div>
        }
      </WhiteButton>
    );
  }
  return (
    <TransparentButton
      style={{ height: '60px' }}
      onClick={onClick}
    >
      {
        props.isLoading
          ? <Spinner id='spinner'/>
          : <div>
            {
              props.imgSrc
                ? <LogoImg
                  src={props.imgSrc}
                />
                : null
            }
            <Text style={{ color: 'white' }}>{props.text}</Text>
          </div>
      }
    </TransparentButton>
  );
}

export default Button;
