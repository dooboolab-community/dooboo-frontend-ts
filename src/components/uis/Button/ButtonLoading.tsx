import type {ReactElement} from 'react';
import styled from '@emotion/styled';

const Spinner = styled.div`
  align-self: center;
  border: 4px solid #f3f3f3; /* Light grey */
  border-top: 4px solid #3498db; /* Blue */
  border-radius: 50%;
  width: 24px;
  height: 24px;
  animation: spin 1s linear infinite;
  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

const ButtonLoading = (): ReactElement => {
  return <Spinner id="spinner" />;
};

export default ButtonLoading;
