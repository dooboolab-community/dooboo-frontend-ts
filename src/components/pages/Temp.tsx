import Button from '../uis/Button';
import {ReactElement} from 'react';
import styled from '@emotion/styled';
import {useNavigate} from 'react-router-dom';

const Container = styled.div`
  flex: 1;
  align-self: stretch;
  justify-content: center;
  align-items: center;
  background: ${(props) => props.theme.background};
  padding: 50px;
`;

function Temp(): ReactElement {
  const navigate = useNavigate();

  return (
    <Container>
      <Button onClick={(): void => navigate(-1)} text="back to tab page" />
    </Container>
  );
}

export default Temp;
