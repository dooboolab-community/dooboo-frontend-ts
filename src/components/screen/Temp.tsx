import React, { Component } from 'react';
import Button from '../shared/Button';
import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  height: 100vh;
  justify-content: center;
  align-items: center;
  background-color: ${(props) => props.theme.mainColor};
  padding: 50px;
`;

interface IProps {
  history?: any;
}

function Temp(props: IProps) {
  const { history } = props;
  return (
    <Container>
      <Button
        onClick={() => history.goBack()}
        text='back to tab page'
      />
    </Container>
  );
}

export default Temp;
