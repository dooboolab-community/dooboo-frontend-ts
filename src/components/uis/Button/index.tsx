import {CSSProperties, ReactElement} from 'react';

import ButtonLoading from './ButtonLoading';
import {ButtonPrimary} from '../Styles';
import styled from '@emotion/styled';

interface Props {
  testID?: string;
  id?: string;
  style?: CSSProperties;
  imgSrc?: string;
  text?: string | ReactElement;
  onClick?: () => void;
  isLoading?: boolean;
}

const ButtonWrapper = styled.div`
  flex: 1;
  height: 100%;

  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;

const PrimaryText = styled.span`
  font-size: 14px;
  color: ${({theme}) => theme.btnPrimaryFont};
  margin: auto;
`;

const LogoImg = styled.img`
  position: absolute;
  left: 20px;
  height: 20px;
  width: 20px;
  object-fit: cover;
`;

function Button(props: Props): ReactElement {
  const {testID, onClick, imgSrc, text, style, isLoading} = props;

  return (
    <ButtonPrimary data-testid={testID} style={style} onClick={onClick}>
      {isLoading ? (
        <ButtonLoading />
      ) : (
        <ButtonWrapper>
          {imgSrc ? <LogoImg src={imgSrc} /> : null}
          <PrimaryText>{text}</PrimaryText>
        </ButtonWrapper>
      )}
    </ButtonPrimary>
  );
}

Button.defaultProps = {
  style: {display: 'flex', height: '60px'},
};

export default Button;
