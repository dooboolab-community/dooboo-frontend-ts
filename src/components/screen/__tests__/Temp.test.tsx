import * as renderer from 'react-test-renderer';

import { fireEvent, getByTestId, render } from '@testing-library/react';

import Button from '../../shared/Button';
import React from 'react';
import Temp from '../Temp';

const props = {
  history: {
    goBack: jest.fn(),
  },
};

describe('[Temp] render', () => {
  it('renders without crashing', () => {
    const rendered = renderer.create(<Temp />).toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });
});

describe('[Temp] Interaction', () => {
  const component = <Temp {...props} />;
  let renderResult: any;

  beforeEach(() => {
    renderResult = render(component);
  });

  it('should simulate [onClick] when [btn] has been clicked', () => {
    const btnInstance = renderResult.getByText('back to tab page');
    fireEvent.click(btnInstance);
    expect(props.history.goBack).toHaveBeenCalledTimes(1);
  });
});
