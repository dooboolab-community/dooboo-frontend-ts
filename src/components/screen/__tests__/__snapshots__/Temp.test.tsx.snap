// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Temp] render renders without crashing 1`] = `
<div
  className="sc-EHOje dSOLHa"
>
  <button
    className="sc-bdVaJa jAKSux"
    onClick={[Function]}
    style={
      Object {
        "height": "60px",
      }
    }
  >
    <div>
      <span
        className="sc-htpNat gUfFlb"
        style={
          Object {
            "color": "white",
          }
        }
      >
        back to tab page
      </span>
    </div>
  </button>
</div>
`;
